packages/styles/src/calendar.scss

```diff
@@ -340,98 +223,190 @@ $fd-calendar-special-days: (
 
-        &--weekend {
-          @extend %fd-calendar-item-weekend;
-        }
+      @include fd-hover() {
+        --fdCalendar_Item_Background: var(--sapList_Hover_Background);
+      }
 
-        &--side-helper {
-          @extend %fd-calendar-item-side-helper;
+      .#{$block}__text {
+        --fdCalendar_Item_Text_Color: var(--sapContent_LabelColor);
+      }
+    }
 
-          .#{$block}__text {
-            color: $fd-calendar-item-helper-color;
-            font-size: $fd-calendar-item-helper-font;
-          }
-        }
+    &--selected {
+      --fdCalendar_Item_Background: var(--sapContent_Selected_Background);
+      --fdCalendar_Item_Border: var(--fdCalendar_Active_Item_Border);
 
-        &--other-month {
-          @extend %fd-calendar-item-other-month;
+      @include fd-hover() {
+        --fdCalendar_Item_Background: var(--fdCalendar_Item_Active_Background_Hover);
+      }
 
-          .#{$block}__text {
-            color: $fd-calendar-item-inactive-color;
-          }
-        }
+      .#{$block}__text {
+        --fdCalendar_Item_Text_Color: var(--sapContent_Selected_TextColor);
 
-        &--range {
-          @extend %fd-calendar-item-range;
+        font-family: var(--fdCalendar_Active_Item_Font_Weight);
+      }
+    }
+
+    &--range {
+      --fdCalendar_Item_Border: var(--fdCalendar_Active_Item_Border);
+      --fdCalendar_Item_Background: var(--sapList_SelectionBackgroundColor);
+
+      @include fd-hover() {
+        --fdCalendar_Item_Background: var(--sapList_Hover_SelectionBackground);
+      }
+
+      // &.#{$block}__item--current { //TODO REVIEW!!!
+      //   @extend %fd-calendar-item-current;
+      // }
+    }
+
+    &--today.#{$block}__item--selected::after {
+      content: "";
+      position: absolute;
+      width: calc(100% - 0.125rem);
+      height: calc(100% - 0.125rem);
+      top: 0.0625rem;
+      bottom: 0.0625rem;
+      left: 0.0625rem;
+      border: solid 0.0625rem var(--sapList_SelectionBorderColor);
+      border-radius: var(--fdCalendar_Item_Today_Selected_Bottom_Radius);
+    }
+
+    &--today.#{$block}__item--range::after {
+      content: "";
+      position: absolute;
+      width: calc(100% - 2px);
+      height: calc(100% - 2px);
+      top: 0.0625rem;
+      bottom: 0.0625rem;
+      left: 0.0625rem;
+      border: solid 0.0625rem var(--sapList_SelectionBorderColor);
+      border-radius: var(--fdCalendar_Item_Today_Selected_Bottom_Radius);
+    }
+
+    &--today {
+      --fdCalendar_Item_Border: solid 0.125rem var(--sapLegend_CurrentDateTime);
+
+      box-shadow: 0 0 0 0.0625rem var(--sapList_Background) inset;
+      position: relative;
 
-          &.#{$block}__item--current {
-            @extend %fd-calendar-item-current;
-          }
+      &::after {
+        content: "";
+        position: absolute;
+        width: calc(100% - 0.125rem);
+        height: calc(100% - 0.125rem);
+        top: 0.0625rem;
+        bottom: 0.0625rem;
+        left: 0.0625rem;
```

Comment:
I see this block of content repeated already 3 times. Maybe extract it to a mixin and reuse it. Will be easier to maintain in future.